Sources/Composed/Providers/ComposedSectionProvider.swift

```diff
@@ -156,18 +160,45 @@ open class ComposedSectionProvider: AggregateSectionProvider, SectionProviderUpd
         insert(child, at: children.count)
     }
 
+    /// Perform multiple updates in a batch, only notifying the delegate that the section provider ended updating
+    /// when all updates have been applied.
+    ///
+    /// It is safe to call this function within a batch updates closure.
+    ///
+    /// - parameter updates: The closure that performs the updates.
+    public func performBatchUpdates(_ updates: (_ sectionProvider: ComposedSectionProvider) -> Void) {
+        if !isPerformingBatchUpdates {
+            isPerformingBatchUpdates = true
+            updateDelegate?.willBeginUpdating(self)
+
+            updates(self)
```

Comment:
I don't know if passing the section provider here is really useful? It might be more confusing than helpful.

Reply:
Yeah I don't think its a good idea tbh. Also definitely not required (for now anyway) so the minimum API would be better IMHO.